scripts/pre_commit_linter.py

```diff
@@ -714,11 +718,142 @@ def _check_spacing(all_files):
     return summary_messages
 
 
+def _check_comments(all_files):
+    """This functions checks for a period at the end of comment."""
+    print 'Starting comment checks'
+    print '----------------------------------------'
+    summary_messages = []
+    files_to_check = [
+        filename for filename in all_files if not
+        any(fnmatch.fnmatch(filename, pattern) for pattern in EXCLUDED_PATHS)
+        and filename.endswith('.py')]
+    message = 'There should be a period at the end of the comment.'
+    failed = False
+    for filename in files_to_check:
+        f = open(filename, 'r')
+        content = f.readlines()
+        length = len(content)
+        line_num = 0
+        while line_num < length:
+            line = content[line_num].lstrip().rstrip()
+            if line.startswith('#'):
+                if line_num + 1 == length:
+                    if line[-1] not in PUNCTUATIONS and line[1:].lstrip()[0].isupper():
```

Comment:
Hmm, I think you're supposed to go for a strict match for periods?

Reply:
There are some cases where there is ``?`` or ``)`` in end of single line comments.

Reply:
But it's not all of them right? (ie. commas, semicolons, colons don't appear to be appropriate to end a comment)

Reply:
So, should I compare with a list of  ``?``, ``)`` or ``.`` then?

Reply:
So, we want to always check for a period in end of single line comments?

Reply:
What should be done with ``?`` or ``)``? Should I change them accordingly?

Reply:
For `)`, that should end in a period.

Do you have a specific case in the codebase where is ends with a question mark? If so, where is it? (This will help me give you the proper guidance :) )

Reply:
In `core/jobs.py` (line 603) we have 
```
# TODO(sll): Need try/except/mark-as-canceled here?
```

Reply:
In that case, let's keep question marks. So the only allowable punctuations are periods and question marks.